src/io/image_olm_parser.rs

```diff
@@ -36,36 +38,91 @@ pub fn render(
+                        .into_iter()
+                        .enumerate()
+                        .fold(Vec::new(), |mut acc, (label, frequency)| {
+                            if frequency > 0 {
+                                let (chunk, _) = chunks.get_index(label).unwrap();
+                                acc.push(chunk.clone())
+                            }
+                            acc
+                        })
+                })
+                // Vec<Vec<DMatrix>> => Vec<Option<DMatrix>>
+                // map to options depending on whether the set of chunks is empty
+                .map(|label_list| label_list.is_empty().not().then(|| label_list))
                 .enumerate()
-                .for_each(|(pixel_index, pixel_alias)| {
-                    let (p_y, p_x) = index_to_coords(pixel_index, chunk_size);
-                    let pixel_y = (top_left_pix_y + p_y) as u32;
-                    let pixel_x = (top_left_pix_x + p_x) as u32;
-                    let pixel = key.get_by_left(pixel_alias).copied().unwrap_or(GREEN);
-                    output_image.put_pixel(pixel_x, pixel_y, pixel);
+                // for each chunk of the image which may contain multiple overlayed opt_chunks
+                .for_each(|(chunk_index, opt_chunks)| {
+                    let (chunk_x, chunk_y) = index_to_coords(chunk_index, graph_width);
+                    // project to pixel coordinates
+                    let top_left_pix_x = chunk_x * chunk_size;
+                    let top_left_pix_y = chunk_y * chunk_size;
+
+                    // Vec<Option<DMatrix>> => Vec<DMatrix>
+                    let chunks: Vec<DMatrix<usize>> = match opt_chunks {
+                        // change an empty set of chunks into a chunk of contradiction pixels
+                        None => vec![DMatrix::from_element(chunk_size, chunk_size, contradiction_key)],
+                        // otherwise returns chunks
+                        Some(chunks) => chunks
+                    };
+
+                    let blend_ratio = 1.0 / chunks.len() as f64;
+
+                    chunks
+                        // a vec of DMatrix of pixel aliases
+                        .iter()
+                        // map each DMatrix of pixel aliases into an iterator rgb values
+                        .map(|chunk| {
+                            chunk
+                                .iter()
+                                .map(|pixel_alias| {
+                                    key
+                                        .get_by_left(pixel_alias)
+                                        .copied()
+                                        .unwrap_or(GREEN)
+                                        // map the pixel to contribute a percentage of the final pixel
+                                        // value proportional to the number of overlayed chunks
+                                        .map(|channel| (channel as f64 * blend_ratio) as u8)
```

Comment:
Unfortuntely the use of `blend_ratio` to compress the pixel value channel won't work. In the case where `chunks.len()` is greater than the value of `channel` the result of the compression will always be 0. This is more clear when we see that `channel * blend_ratio == channel / chunks.len()`, and then remembering that float into uint always floors the value. Since with OLM we've already seen instances of over 300 chunks it's likely we'll encounter this scenario again. I would suggest just doing the division later after summing the values (casting up to `usize` or whatever).